errors.go

```diff
@@ -58,3 +58,15 @@ type marshalable interface {
 	json.Marshaler
 	json.Unmarshaler
 }
+
+// Error wraps RPC errors, which contain an error code in addition to the message.
+type Error interface {
```

Comment:
Why do we need this interface ?

Reply:
Added so we can have custom error codes.

Reply:
Why do we need custom error codes ? Can you write a bit explaining what made you add this ? Was it something you saw during your testing ?

Reply:
Also this needs a better name. This can be `ErrorWithCode` and the below can be `ErrorWithData`

Reply:
@aarshkshah1992 I think this will give more idea towards the error codes: https://github.com/filecoin-project/lotus/pull/12553/#discussion_r1800668415